### PR TITLE
Replace header guards in API with pragma once

### PR DESCRIPTION
**Pull request by Harrietbrown**
This PR replaces all header guards in API with #pragma once

**To test:**

Any issues in this PR should be caught by jenkins.

This pr is a part of #28200

*This does not require release notes* because it is an internal change to the codebase.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/API/test/FileFinderTest.h [hunk 1]
```diff
@@ -471,7 +470,6 @@ class FileFinderTest : public CxxTest::TestSuite {
     TS_ASSERT(path.find("CSp78173.Raw") != std::string::npos);
 #else
     TS_ASSERT(path.find("CSP78173.raw") != std::string::npos);
-#endif
```
Comment (StephenSmith25) [inline Framework/API/test/FileFinderTest.h]: I think this the replace script is removing too many #endif macros

Comment (Harrietbrown) [reply Framework/API/test/FileFinderTest.h]: I thaught i caught that one, the script is still in progress.

**Comment by StephenSmith25** [comment]
It appears you've missed the .tcc files in this directory, I'm assuming these should be changed to #pragma once as well, but may need confirmation on this. As an example, Algorithm.tcc

**Comment by Harrietbrown** [comment]
Good catch, I just checked, I'll add tcc files to my replacer.

**Review by StephenSmith25 — APPROVED** [review]
Looks good to go :+1:

**Pull request merged**